### PR TITLE
Add exitscript object

### PR DESCRIPTION
**Pull request by ahmad-shahid**
Addresses issue #51

### CODE REVIEW

File: src/gcweb/refFooter.ejs [hunk 1]
```diff
@@ -22,8 +23,14 @@
 <script src="<%- wet.builder.environment(pr.cdnEnv) %>wet-boew/js/ie8-wet-boew2.min.js"></script>
 <![endif]-->
 <script src="<%- wet.builder.environment(pr.cdnEnv) %>wet-boew/js/theme.min.js"></script>
-<%_ if (pr.exitScript == true) { -%>
-    <%# This is a placeholder link to trigger the WET exitscript intialization in the event the no external link is present on the page -%>
+<%_ if (pr.exitSecureSite != null) { -%>
+    <%_ if (pr.exitSecureSite.exitScript == true) { -%>
+        <%# This is a placeholder link to trigger the WET exitscript intialization in the event the no external link is present on the page -%>
+        <a href="#" id="cdts-exitscript-placeholder-link" style="display: none" aria-hidden="true" class="wb-exitscript">Exit Link</a>
+        <script data-id="exitScript">cdtsApplyExitScriptToLinks("<%=pr.exitSecureSite.displayModal%>", "<%-pr.exitSecureSite.exitURL%>", "<%=pr.exitSecureSite.exitDomains%>", "<%=pr.exitSecureSite.exitMsg%>", "<%=pr.exitSecureSite.yesMsg%>", "<%=pr.exitSecureSite.cancelMsg%>", "<%=pr.exitSecureSite.msgBoxHeader%>", "<%=pr.exitSecureSite.targetWarning%>", "<%=pr.exitSecureSite.displayModalForNewWindow%>");</script>
```
Comment (plup2) [inline src/gcweb/refFooter.ejs:30]: The statements `param != null? param: 'undefined'` were removed but the `cdtsApplyExitScriptToLinks` function is still expecting undefined values to be `"undefined"` and not `undefined`.   Unless the function itself is modified to change that expectation, all the `?` operations will have to be kept.  (In other words, The difference between the backwards compatible line and this one should only be the addition of `.exitSecureSite` for each parameter.

(same goes for gcintranet file)

**Review by plup2 — CHANGES_REQUESTED** [review]

File: src/gcweb/refFooter.ejs (hunk 1 above)
Comment (ahmad-shahid) [reply src/gcweb/refFooter.ejs:30]: In the exitscript, there is a condition to check if the value is a string and return undefined.
`if (value === "undefined") {
          return undefined;
        }
        return value;`

Would that be ok or do we need to make this change?

Comment (plup2) [reply src/gcweb/refFooter.ejs:30]: That check is fine but only affects the object that will go in the HTML element.

The code still does this:
```javascript
...
    if (targetWarning === "undefined") targetWarning = " ";
...
```

and this 

```javascript
...
if (exitUrl !== "undefined"){
                    links[i].href = encodeURI(exitUrl + (exitUrl.indexOf('?') < 0 ? "?": "&") + "targetUrl=") + encodeURIComponent(links[i].href);
}
...
```

So `"undefined"` (and not `undefined`) still needs to be passed

Comment (ahmad-shahid) [reply src/gcweb/refFooter.ejs]: That makes sense. It's fixed.

**Review by plup2 — APPROVED** [review]

**Pull request merged**